### PR TITLE
Fix some broken ZipFile unicode tests

### PR DESCRIPTION
**Pull request by ianhays**
These tests rely on the use of the FileData class to verify zips are created/extracted with the correct files. However, FileData and the accompanying helper methods do not take unicode normalization into account, so unicode zip tests are failing on OSX. I modified these tests to verify based on the actual files being zipped/unzipped instead of using the expectations set in the FileData class.

resolves https://github.com/dotnet/corefx/issues/10397, https://github.com/dotnet/corefx/issues/9867

@stephentoub

### CODE REVIEW

**Comment by ianhays** [comment]
Fixes these three failing tests on [mc.dotnet](https://mc.dot.net/#/product/netcore/master/source/official~2Fcorefx~2Fmaster/type/test~2Ffunctional~2Fcli~2F/build/20160901.02/workItem/System.IO.Compression.ZipFile.Tests)

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs [hunk 1]
```diff
@@ -15,28 +17,29 @@ public class ZipFileTest_ConvenienceMethods : ZipFileTestBase
...
+            string folderName = zfolder("normal");
+            string noBaseDir = GetTestFilePath();
+            ZipFile.CreateFromDirectory(folderName, noBaseDir);
 
-        [Fact]
-        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Jenkins fails with unicode characters [JENKINS-12610]
-        public async Task CreateFromDirectoryUnicodel()
-        {
-            await TestCreateDirectory(zfolder("unicode"), true);
+            await IsZipSameAsDirAsync(noBaseDir, folderName, ZipArchiveMode.Read, true, true);
```
Comment (stephentoub) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: Nit: can you name the Boolean args so it's clear what each of the ```true```s mean?  Same for other cases of the same thing.

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs [hunk 2]
```diff
@@ -15,28 +17,29 @@ public class ZipFileTest_ConvenienceMethods : ZipFileTestBase
...
+
+            string withBaseDir = GetTestFilePath();
```
Comment (stephentoub) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: This seems like it's a completely separate test from the one above... is it?  Consider putting it into its own test method.

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs [hunk 3]
```diff
@@ -67,33 +70,42 @@ private static void SameExceptForBaseDir(string zipNoBaseDir, string zipBaseDir,
...
+            }
         }
 
         [Fact]
-        [Trait(XunitConstants.Category, XunitConstants.IgnoreForCI)] // Jenkins fails with unicode characters [JENKINS-12610]
-        public void ExtractToDirectoryUnicode()
+        public void ExtractToDirectoryNull()
         {
-            TestExtract(zfile("unicode.zip"), zfolder("unicode"));
+            Assert.Throws<ArgumentNullException>(() => ZipFile.ExtractToDirectory(null, GetTestFilePath()));
```
Comment (stephentoub) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: Nit: is there an argument name to verify?

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs [hunk 4]
```diff
@@ -67,33 +70,42 @@ private static void SameExceptForBaseDir(string zipNoBaseDir, string zipBaseDir,
...
+            string folderName = zfolder("unicode");
             using (var tempFolder = new TempDirectory(GetTestFilePath()))
             {
                 ZipFile.ExtractToDirectory(zipFileName, tempFolder.Path);
-                DirsEqual(tempFolder.Path, folderName);
-
-                Assert.Throws<ArgumentNullException>(() => ZipFile.ExtractToDirectory(null, tempFolder.Path));
+                IEnumerable<string> actual = Directory.EnumerateFileSystemEntries(tempFolder.Path, "*", SearchOption.AllDirectories);
+                IEnumerable<string> expected = Directory.EnumerateFileSystemEntries(folderName, "*", SearchOption.AllDirectories);
+                Assert.True(Enumerable.SequenceEqual(expected.Select(i => Path.GetFileName(i)), actual.Select(i => Path.GetFileName(i))));
```
Comment (stephentoub) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: This looks similar to code I saw earlier in the PR.  Is it the same?  Should it be made it into a helper and reused?

**Comment by stephentoub** [comment]
A few questions/comments, but otherwise LGTM.  Thanks.

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs (hunk 2 above)
Comment (ianhays) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: It certainly can be.  I'll move it to its own method and combine it with that helper that isn't used anywhere else.

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs (hunk 4 above)
Comment (ianhays) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: These three lines are used twice in this file. The usage earlier in the `CreateFromDirectoryUnicode` test is similar but not enough to cleanly parameterize with this one.

I'll move these to a helper method though.

File: src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs (hunk 1 above)
Comment (ianhays) [inline src/System.IO.Compression.ZipFile/tests/ZipFileConvenienceMethods.cs]: Done. I removed the unnecessary negative and updated it for Compression as well as Compression.ZipFile.

**Comment by ianhays** [comment]
test Innerloop CentOS7.1 Debug Build and Test please

**Pull request merged**